search/util.py

```diff
@@ -0,0 +1,16 @@
+"""Utility functions for search"""
+
+
+def traverse_mapping(mapping):
+    """
+    Traverse a mapping, yielding each node
+
+    Args:
+        mapping (dict): A dictionary within the mapping
```

Comment:
a couple notes for this docstring:

1. I think the first line should read "Traverse a mapping, yielding each nested dict"
1. I think this line should read "mapping (dict): The mapping itself, or a nested dict"